src/nidcpower/system_tests/test_system_nidcpower.py

```diff
@@ -1066,9 +1066,28 @@ def session_creation_kwargs(self):
         return {}
 
 
-class TestGrpc(SystemTests):
-    server_address = "localhost"
-    server_port = "31763"
+class GrpcServerProcess:
+    def __init__(self):
+        server_exe = self._get_grpc_server_exe()
+        self._proc = subprocess.Popen([str(server_exe)], stdout=subprocess.PIPE)
+
+        # Read/parse first line of output; discard the rest
+        try:
+            first_line = self._proc.stdout.readline()
+            assert first_line.startswith(b"Server listening on port "), f"Unrecognized output: {first_line}"
+            self.server_port = int(first_line.replace(b"Server listening on port ", b"").strip())
+
+            self._stdout_thread = threading.Thread(target=self._discard_output, args=(self._proc.stdout,), daemon=True)
+            self._stdout_thread.start()
```

Comment:
We're doing this just to close the process more cleanly, it looks like?

Reply:
If the process prints a bunch of stuff to STDOUT and fills up its output buffer, it'll hang on the next print if we don't empty it. It is not a problem currently, but I added it to be safe.